File: RepoTransBench/target_projects/Python/C++/mapio_GraphvizAnim/public_tests/test_public_action_extra.cpp
#include <gtest/gtest.h>
#include "action.h"
#include <vector>

TEST(PublicActionExtraTest, AddNodeActionPublic) {
    std::vector<Step> s{Step()};
    AddNode(12)(s);
    EXPECT_NE(s.back().V.find(12), s.back().V.end());
}

TEST(PublicActionExtraTest, HighlightNodeAndLabelNodePublic) {
    std::vector<Step> s{Step()};
    HighlightNode(20, "red")(s);
    LabelNode(20, "lbl2")(s);
    EXPECT_NE(s.back().hV.find(20), s.back().hV.end());
    EXPECT_NE(s.back().lV.find(20), s.back().lV.end());
}

TEST(PublicActionExtraTest, UnlabelAndRemoveNodePublic) {
    std::vector<Step> s{Step()};
    AddNode(9)(s);
    LabelNode(9, "zzz")(s);
    UnlabelNode(9)(s);
    EXPECT_EQ(s.back().lV.find(9), s.back().lV.end());
    RemoveNode(9)(s);
    EXPECT_EQ(s.back().V.find(9), s.back().V.end());
}

TEST(PublicActionExtraTest, AddEdgeAndHighlightLabelUnlabelRemovePublic) {
    std::vector<Step> s{Step()};
    AddNode(6)(s);
    AddNode(13)(s);
    AddEdge(6, 13)(s);
    HighlightEdge(6, 13, "purple")(s);
    LabelEdge(6, 13, "Y")(s);
    EXPECT_NE(s.back().E.find({6, 13}), s.back().E.end());
    EXPECT_NE(s.back().hE.find({6, 13}), s.back().hE.end());
    EXPECT_NE(s.back().lE.find({6, 13}), s.back().lE.end());
    UnlabelEdge(6, 13)(s);
    RemoveEdge(6, 13)(s);
    EXPECT_EQ(s.back().E.find({6, 13}), s.back().E.end());
}

File: RepoTransBench/target_projects/Python/Rust/mapio_GraphvizAnim/public_tests/test_public_action_extra.rs
use gvanim::action;
use gvanim::animation::Step;

#[test]
fn test_add_node_action_public() {
    let mut s = vec![Step::default()];
    action::AddNode(12).call(&mut s);
    assert!(s.last().unwrap().V.contains(&12));
}

#[test]
fn test_highlight_node_and_label_node_public() {
    let mut s = vec![Step::default()];
    action::HighlightNode { node: 20, color: "red".to_string() }.call(&mut s);
    action::LabelNode { node: 20, label: "lbl2".to_string() }.call(&mut s);
    let st = s.last().unwrap();
    assert!(st.hV.contains_key(&20) && st.lV.contains_key(&20));
}

#[test]
fn test_unlabel_and_remove_node_public() {
    let mut s = vec![Step::default()];
    action::AddNode(9).call(&mut s);
    action::LabelNode { node: 9, label: "zzz".to_string() }.call(&mut s);
    action::UnlabelNode(9).call(&mut s);
    assert!(!s.last().unwrap().lV.contains_key(&9));
    action::RemoveNode(9).call(&mut s);
    assert!(!s.last().unwrap().V.contains(&9));
}

#[test]
fn test_add_edge_and_highlight_label_unlabel_remove_public() {
    let mut s = vec![Step::default()];
    action::AddNode(6).call(&mut s);
    action::AddNode(13).call(&mut s);
    action::AddEdge(6, 13).call(&mut s);
    action::HighlightEdge { n1: 6, n2: 13, color: "purple".to_string() }.call(&mut s);
    action::LabelEdge { n1: 6, n2: 13, label: "Y".to_string() }.call(&mut s);

    let st = s.last().unwrap();
    assert!(st.E.contains(&(6, 13)) && st.hE.contains_key(&(6, 13)) && st.lE.contains_key(&(6, 13)));
    action::UnlabelEdge(6, 13).call(&mut s);
    action::RemoveEdge(6, 13).call(&mut s);
    assert!(!s.last().unwrap().E.contains(&(6, 13)));
}